res/controllers/Numark-Mixtrack-3-scripts.js

```diff
@@ -148,25 +149,44 @@ var intervalsPerRev = 1200,
  *            - removed trailing empty lines at end of script
  *            - line 1749, change .75 to 0.75 in var gammaOutputRange 
  *            - added spacing in numerous place for easier reading
- *2016-09-14 (1.3 ) - Stefan Mikolajczyk - https://github.com/mixxxdj/mixxx/pull/1012
+ *2016-09-14 (1.31 ) - Stefan Mikolajczyk - https://github.com/mixxxdj/mixxx/pull/1012
  *            - changed PADLoopButton behaviour to fit native serato behaviour (2, 4, 8, 16 instead of 1/8, 1/4, 1/2, 1)
+ *2016-04-08 (1.4 ) - Stéphane Morin -
+ *            - Adjusted mapping of AutoLoop to be same as Serato and Virtual DJ, 
+ *              1,2,4,8 if not shifted, previously 0.125, 0.25, 0.5, 16 if shifted
+ *2016-08-31 (1.5 ) - Stéphane Morin -
+ *            - Bug Touch platter + wheel touch speeds up/slows down song 
+ *              when wheel not enabled (WheelMove - Jog / Pitch bend)
+ *            - Changed mapping of FX sections: 
+ *              -Strip = Parameter 1, TAP + Strip = Super knob
+ *              -Beat knob = parameter 2; TAP + Beat knob = Mix ratio
+ *              -PadModeButton + Beat knob = Sampler volumes
+ *              -Shift + BeatKnob = Beat grid move
+ *              -Correct Tap button led behavior when shift lock is active
+ *              -Remove Tap button BPM Tap mapping 
+ *2016-09-06 (1.6 ) - Stéphane Morin -
+ *              -Corrected Tap button mapping
+ *
+ *2016-09-06 (1.7 ) - Stéphane Morin - FX section rework
+ *              - Connected controls for FX buttons 
+ *              - Removed soft takeover for Super FX knob
+ *              - Removed Callback function for FX Loaded to prevent all 
+ *              - parameters from being automatically linked to Super Knob
+ *              - Added InstantFx 
+ *              - removed configuration option BeatKnobAsSamplerVolume
+  *2016-09-20 (1.8 ) - Stéphane Morin - FX parameter rework
+ *              - Addition of EQ buttons in script in order to also use them for adjusting FX parameters
+ *              - Added Parameter soft takeover function
+ *              - corrected smartPFL callback for when loading track from UI.
  *
  * To do - (maybe..)
  * ----------------
- *            - Add script to control "volume","filterHigh","filterMid","filterLow" and"crossfader"
- *              In order for engine.softTakeover to work for these controls
+ *            - Add script to control "volume" and"crossfader"
+ *              in order for engine.softTakeover to work for these controls
```

Comment:
> You can probably merge NumarkMixtrack3.TrebleKnob NumarkMixtrack3.MiddleKnob and NumarkMixtrack3.BassKnob into one function to avoid code duplication

How would you do it?